unit_tests/speech/test_client.py

```diff
@@ -0,0 +1,111 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+
+
+class TestClient(unittest.TestCase):
+
+    def _getTargetClass(self):
+        from google.cloud.speech.client import Client
+        return Client
+
+    def _makeOne(self, *args, **kw):
+        return self._getTargetClass()(*args, **kw)
+
+    def test_ctor(self):
+        from google.cloud.speech.connection import Connection
+
+        creds = _Credentials()
+        http = object()
+        client = self._makeOne(credentials=creds, http=http)
+        self.assertTrue(isinstance(client.connection, Connection))
+        self.assertTrue(client.connection.credentials is creds)
+        self.assertTrue(client.connection.http is http)
+
+    def test_syncrecognize(self):
+        from google.cloud.speech.client import Encoding
+
+        creds = _Credentials(object())
+        http = _Http(headers={},
+                     content={
+                         "results": [
+                             {
+                                 "alternatives": [
+                                     {"transcript": "hello",
+                                      "confidence": 0.784919}
+                                     ]
+                             }
+                         ]
+                     })
+        client = self._makeOne(credentials=creds, http=http)
+
+        uri = 'gs://a-bucket/rec_sample.flac'
+        encoding = Encoding.FLAC
+        sample_rate = 16000
+        hints = ["test"]
+        speechrecognition_result = client.syncrecognize(None, uri, encoding,
+                                                        sample_rate,
+                                                        max_alternatives=2,
+                                                        speech_context=hints)
+        self.assertEqual(speechrecognition_result[0]["transcript"], 'hello')
+
+        b64_speech = "binarycontent"
+        speechrecognition_result = client.syncrecognize(b64_speech, None,
+                                                        encoding,
+                                                        sample_rate,
+                                                        max_alternatives=2)
+        self.assertEqual(speechrecognition_result[0]["transcript"], 'hello')
+
+    def test_syncrecognize_failure(self):
+        creds = _Credentials()
+        client = self._makeOne(credentials=creds)
+
+        with self.assertRaises(ValueError):
+            client.syncrecognize("content", "uri", None, None)
+        with self.assertRaises(ValueError):
+            client.syncrecognize(None, None, None, None)
+        with self.assertRaises(ValueError):
+            client.syncrecognize(None, "uri", None, None)
```

Comment:
Please split each failure path out into its own testcase, with a method name which describes it.